testing/proxytest/proxytest.go

```diff
@@ -143,6 +154,32 @@ func New(t *testing.T, optns ...Option) *Proxy {
 	return &p
 }
 
+func (p *Proxy) Start() {
+	p.Server.Start()
+	u, err := url.Parse(p.URL)
+	if err != nil {
+		panic(fmt.Sprintf("could parse fleet-server URL: %v", err))
```

Comment:
Why not make this return an error and let the caller handle it?

Reply:
Originally the Start() and StartTLS() methods were not defined for the `Proxy` struct as it exposed the ones from `*httptest.Server`. It wasn't even used in tests as the server was created and started in one go.
Now that we are overriding those methods we can return what we want, will add a small commit for that

Reply:
Done in 979217c0b27b525e39d2344354e5e2e6363198b4